internal/uvm/virtual_device.go

```diff
@@ -6,20 +6,41 @@ import (
 
 	"github.com/Microsoft/go-winio/pkg/guid"
 	"github.com/Microsoft/hcsshim/internal/guestrequest"
+	"github.com/Microsoft/hcsshim/internal/log"
 	"github.com/Microsoft/hcsshim/internal/requesttype"
 	hcsschema "github.com/Microsoft/hcsshim/internal/schema2"
 )
 
-func (uvm *UtilityVM) AssignDevice(ctx context.Context, device hcsschema.VirtualPciDevice) (string, error) {
+// VPCIDevice represents a vpci device. Holds its guid and a handle to the uvm it
+// belongs to.
+type VPCIDevice struct {
+	vm *UtilityVM
+	ID string
```

Comment:
I'd rather have a getter function on VPCIDevice for getting the ID than having it be exported, but I'm not sure it really matters. I guess just for consistency and showing more explicitly that ID is used outside the package.

Reply:
We already spoke on this but for now to be consistent with the others I'll leave it exported instead of a getter.